src/Microsoft.Extensions.FileProviders.Abstractions/NullFileProvider/NullFileInfo.cs

```diff
@@ -0,0 +1,33 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+
+namespace Microsoft.Extensions.FileProviders.Abstractions
+{
+    public class NullFileInfo : IFileInfo
+    {
+        public NullFileInfo(string name)
+        {
+            Name = name;
+        }
+
+        public bool Exists => false;
+
+        public bool IsDirectory => false;
+
+        public DateTimeOffset LastModified => DateTimeOffset.MinValue;
+
+        public long Length => -1;
+
+        public string Name { get; }
+
+        public string PhysicalPath => null;
+
+        public Stream CreateReadStream()
+        {
+            throw new FileNotFoundException(string.Format($"{nameof(NullFileProvider)} does not support reading files.", Name));
```

Comment:
Good point. In fact I think the code where I copied this from originally intended the `Name` to be the second argument of `FileNotFoundException()`